src/cats/BreedSelector.tsx

```diff
@@ -0,0 +1,66 @@
+import { useCallback, useMemo } from "react";
+import clsx from "clsx";
+import { Status } from "./useAsync";
+import styles from "./BreedSelector.module.css";
+
+type BreedType = {
+  id: string;
+  name: string;
+};
+
+type BreedSelectorPropsType = {
```

Comment:
nit: always name the props for the main component `Props`. It is a good signal and the main thing people are looking for when they open a component they might want to use.

Less important, but we recently switched to using `interface` everywhere as there is a performance benefit (`interface` is opaque to the typechecker so it does less work, whereas `type` is always expanded). That being said I prefer the `type` syntax from using Flow for so long, so don't put much stock in this.

Reply:
The performance benefit is only at compile time, it would make no difference at runtime, correct? I'm surprised it's substantial enough to warrant switching.

Do you still end up using types for discriminated unions, e.g:

```
interface Cash {
kind: "cash";
}

interface PayPal {
kind: "paypal";
email: string;
}

interface CreditCard {
kind: "creditcard";
cardNumber: string;
securityCode: string;
}

type PaymentMethod = Cash | PayPal | CreditCard;
```

Reply:
Yes, its only at typecheck time that you gain perf. We saw a 20% speedup in our codebase by converting a bunch of widely used core types to `interface`. And yes, types for unions is still necessary (part of the reason I prefer type everywhere).